### PR TITLE
Add bandit linter

### PR DESCRIPTION
**Pull request by MartinBasti**
Bandit is linter for detecting potential security issues:
* replaced XML parser with safe parser defusedxml in pip.py
* added "# nosec" in many places as they were just false positive alerts
* added bandit to tox
* added bandit to GH workflow

Signed-off-by: Martin Basti <mbasti@redhat.com>

### CODE REVIEW

**Comment by tkdchen** [comment]
:+1:

**Review by serg-cymbaluk — APPROVED** [review]
LGTM

**Review by brunoapimentel — APPROVED** [review]

**Comment by brunoapimentel** [comment]
LGTM :+1:

File: cachito/workers/config.py [hunk 1]
```diff
@@ -163,7 +163,7 @@ def configure_celery(celery_app):
         # a hack taken from flask.config.from_pyfile.
         _user_config = {}
         with open(prod_config_file_path, mode="rb") as config_file:
-            exec(compile(config_file.read(), prod_config_file_path, "exec"), _user_config)
+            exec(compile(config_file.read(), prod_config_file_path, "exec"), _user_config)  # nosec
```
Comment (akhmanova) [inline cachito/workers/config.py:166]: I will be glad to see a comment on why we want `bandit` to ignore this line. And maybe a comment for one of the `password` lines.

**Comment by akhmanova** [comment]
Lol, sorry, I didn't reload this page.

**Pull request merged**